kernel/src/process/signal/poll.rs

```diff
@@ -82,11 +152,24 @@ impl Pollee {
     ///
     /// This method wakes up all registered pollers that are interested in the events.
     ///
-    /// The events can be spurious. This way, the caller can avoid expensive calculations and
-    /// simply add all possible ones.
+    /// This method should be called whenever new events arrive. The events can be spurious. This
+    /// way, the caller can avoid expensive calculations and simply add all possible ones.
     pub fn notify(&self, events: IoEvents) {
+        self.invalidate();
+
         self.inner.subject.notify_observers(&events);
     }
+
+    /// Invalidates old events.
```

Comment:
I think using the verb _invalidate_ with the word _cached_ would make the name more self-describing.

```
/// Invalidates the (internal) cached events.
```